src/libraries/System.IO.FileSystem/tests/Enumeration/AttributeTests.cs

```diff
@@ -136,5 +159,29 @@ public void IsHiddenAttribute()
 
             Assert.Equal(new string[] { fileTwo.FullName }, enumerable);
         }
+
+        [Fact]
+        public void IsReadOnlyAttribute()
+        {
+            DirectoryInfo testDirectory = Directory.CreateDirectory(GetTestFilePath());
+
+            FileInfo fileOne = new FileInfo(Path.Combine(testDirectory.FullName, GetTestFileName()));
+            FileInfo fileTwo = new FileInfo(Path.Combine(testDirectory.FullName, GetTestFileName()));
+
+            fileOne.Create().Dispose();
+            fileTwo.Create().Dispose();
+
+            fileTwo.Attributes |= FileAttributes.ReadOnly;
+
+            IEnumerable<string> enumerable = new FileSystemEnumerable<string>(
+                 testDirectory.FullName,
+                 (ref FileSystemEntry entry) => entry.ToFullPath(),
+                 new EnumerationOptions() { AttributesToSkip = 0 })
```

Comment:
Yes, the IsHiddenAttribute method starts failing if I don't explicitly set this to 0 (to signify "don't skip any attributes").

The reason is because when unset, then by default we skip the `Hidden` and the `System` attributes. That's how `EnumerationOptions.AttributesToSkip` is documented.

> Skip entries with the given attributes. Default is FileAttributes.Hidden | FileAttributes.System.